source/code/plugins/out_oms_api.rb

```diff
@@ -164,10 +164,19 @@ def write(chunk)
       # Group records based on their datatype because OMS does not support a single request with multiple datatypes.
       datatypes = {}
       chunk.msgpack_each {|(tag, record)|
-        if !datatypes.has_key?(tag)
-          datatypes[tag] = []
+        if !record.to_s.empty?
+          if !datatypes.has_key?(tag)
+            datatypes[tag] = []
+          end
+
+          if record.is_a?(Array)
+            record.each do |r|
+              datatypes[tag] << r if !r.to_s.empty? and r.is_a?(Hash)
+            end
+          elsif record.is_a?(Hash)
```

Comment:
Yes, the service side expects a JObject or a JArray of JObject, but it cannot handle a JArray of JArray. Since we batch process the records, we need to make sure it always send a JArray of JObject. No simple type is allowed.